### PR TITLE
Merge master into 1.17

### PR DESCRIPTION
**Pull request by wagyourtail**

### CODE REVIEW

File: src/main/java/baritone/cache/WorldProvider.java [hunk 1]
```diff
@@ -75,7 +75,18 @@ public final void initWorld(ResourceKey<Level> worldKey, DimensionType world) {
...
+            if (mc.isConnectedToRealms()) {
+                folderName = "realms";
+            } else {
+                if (mc.getCurrentServer() != null) {
+                    folderName = mc.getCurrentServer().ip;
+            } else {
+                //replaymod causes null currentServerData and false singleplayer.
+                currentWorld = null;
+                return;
+            }
```
Comment (scorbett123) [inline src/main/java/baritone/cache/WorldProvider.java:88]: This was probably an issue that I should have noticed on the initial commit but the indentation here is really confusing, should you fix it in this pr, or should I open a new one to fix it?

**Review by scorbett123 — APPROVED** [review]
lgtm

File: src/main/java/baritone/cache/WorldProvider.java (hunk 1 above)
Comment (wagyourtail) [reply src/main/java/baritone/cache/WorldProvider.java:88]: o oops. no that's an issue with indenting here specifically... I fix

**Review by ZacSharp — APPROVED** [review]
I think we should still merge via the other branches for major mc versions (and I know my pr is outdated). However that's nothing that can't be done after merging this and this merge is really needed.

**Comment by ZacSharp** [comment]
Just saw the deprecation commits, forget what I said about older versions.

**Comment by wagyourtail** [comment]
yeah, this is how leij told me to do it in discord

**Comment by ZacSharp** [comment]
Nope, no discord at all.

**Pull request merged**